### DISCUSSION TITLE
Is it default behaviour to load `conftest.py` and set up fixtures when calling `--help`?

### DISCUSSION THREAD

**Discussion by daria-shaw** [pytest-dev/pytest#12718] [Q&A] [answered]
I have some slow to setup fixtures that live in `tests/conftest.py`. When I run `pytest --help`, it takes a little time before it returns anything to the terminal, but if I run `pytest --help --noconftest` then the output is instantaneous.

Is it expected behaviour to load in conftest when help is invoked?

**Comment by RonnyPfannschmidt** [comment, marked as answer]
if your fixtures set up when a conftest is loaded, then its not a fixture, its a cursed import time global and you should rethink your choices

conftest loading is expected as conftests can add cli args

fixtures are not set up until after collecting all tests

**Reply by daria-shaw** [reply]
defo a me issue loading or importing something I shouldn't in a test file then.

is there an easy method of checking what's getting loaded/collected while running `pytest --help`?

**Reply by RonnyPfannschmidt** [reply]
there is no collection for help printing

please show the conftest
